pkg/scheduler/controller/shoot/reconciler.go

```diff
@@ -59,16 +59,6 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 
 	logger := r.logger.WithValues("shoot", client.ObjectKeyFromObject(shoot))
 
-	if shoot.Spec.SeedName != nil {
-		logger.Info("Shoot already scheduled onto seed, nothing left to do", "seed", *shoot.Spec.SeedName)
-		return reconcile.Result{}, nil
-	}
-
-	if shoot.DeletionTimestamp != nil {
-		logger.Info("Ignoring shoot because it has been marked for deletion")
-		return reconcile.Result{}, nil
-	}
```

Comment:
I think, this shouldn't be removed, although it is already checked in the predicates. The reason is, that predicates are only evaluated for watch events, but not when an object is requeued.
So if a shoot is in exponential backoff it might get scheduled manually or marked as deleted in between retries.